content/chat/getting-started/javascript.textile

```diff
@@ -0,0 +1,195 @@
+```
+
+* Use the Ably CLI to send a message to your first room: @ably rooms messages send my-first-channel 'Hello!'@. The message will be received by the client you've subscribed to the room, and be logged to the console.
+
+h2(#step-3). Step 3: Edit a message
+
+If your client makes a typo, or needs to update their original message then they can edit it.
+
+* Add the following line to your @getStarted()@ function to edit your original first message. Technically this will be your second message that you edit, as running the function again will re-send the original message.
+
+```[javascript]
+await room.messages.update(myFirstMessage.copy( {text: 'My 1st message! (edited)'}));
+```
+
+
+h2(#step-4). Step 4: Show who is typing a message
+
+Typing indicators enable you to display messages to clients when someone is currently typing. An event is emitted when someone starts typing, when they press a keystroke, and then another event is emitted after a configurable amount of time has passed without a key press.
+
+* In practice the @typing.start()@ method would be called on keypress, however for demonstration purposes we will call it and wait for the default period of time to pass for a stop event to be emitted. Add the following lines to @getStarted@ to subscribe to typing events and emit one, then run it:
+
+```[javascript]
+const {unsubscribe: typingUnsubscribe} = room.typing.subscribe((typingEvent) => {
+  {
+    if (typingEvent.currentlyTyping.size === 0) {
+      console.log('No one is currently typing');
+    } else {
+      console.log(`${[...typingEvent.currentlyTyping][0]} is typing`);
+    }
+  }
+});
+
+await room.typing.start()
+```
+
+* Use the Ably CLI to subscribe to typing events: @ably rooms typing subscribe my-first-room@. You will see that the client in your IDE starts to type after re-running @getStarted()@, with the message @my-first-client is typing...@.
+
+h2(#step-5). Step 5: Display online status
+
+Display the online status of clients using the presence feature. This enables clients to be aware of one another if they are present in the same room. This enables you to show clients who else is online, provide a custom status update for each, and notify the room when someone enters it, or leaves it, such as by going offline.
+
+* Add the following lines to your @getStarted()@ function to subscribe to, and join, the presence set of the room. It also sets a short wait before leaving the presence set. Run @getStarted()@ after updating it:
+
+```[javascript]
+const {unsubscribe: presenceUnsubscribe} = room.presence.subscribe((member) => {
+    console.log(`Event type: ${member.action} from  ${member.clientId} with the data ${member.data}`)
+});
+
+await room.presence.enter(`I'm here!`);
+
+setTimeout( () => {
+  room.presence.leave(`I'm leaving!`);
+}, 3000);
+```
+
+* You can have another client join the presence set using the Ably CLI: @ably rooms presence enter my-first-room --client-id "my-cli"@.
```

Comment:
`Event type: enter from  my-cli with the data [object Object]`

This is what the console says. Not sure if it's intended to show `[object Object]`